TypeScript Error `TS2590: Expression produces a union type that is too complex to represent.`

Since Execa 9.0.0, when using TypeScript 5.1 or later, the following type error appears:

```
types/subprocess/stdout.d.ts:18:4 - error TS2590: Expression produces a union type that is too complex to represent.

18  : InputOutputStream<IsInputFd<FdNumber, OptionsType>>;
```

Or:

```
types/return/result-stdout.d.ts:33:4 - error TS2590: Expression produces a union type that is too complex to represent.

 33  : ResultStdioItem<
 34  IsObjectFd<ObjectFdNumber, OptionsType>,
... 
 36  OptionsType['encoding']
 37  >;
```

Apparently the error is not emitted when using `tsd`, only when using `tsc`, which is why our type tests did not catch this.

--- a/types/transform/object-mode.d.ts
+++ b/types/transform/object-mode.d.ts
@@ -1,7 +1,7 @@
 import type {StdioSingleOption, StdioOptionCommon} from '../stdio/type';
 import type {FdStdioOption} from '../stdio/option';
 import type {CommonOptions} from '../arguments/options';
-import type {GeneratorTransformFull, DuplexTransform, WebTransform} from './normalize';
+import type {DuplexTransform} from './normalize';
 
 // Whether a file descriptor is in object mode
 // I.e. whether `result.stdout|stderr|stdio|all` is an array of `unknown` due to `objectMode: true`
@@ -15,14 +15,10 @@ type IsObjectStdioOption<StdioOptionType extends StdioOptionCommon> = IsObjectSt
 	: StdioOptionType
 >;
 
-type IsObjectStdioSingleOption<StdioSingleOptionType extends StdioSingleOption> = StdioSingleOptionType extends GeneratorTransformFull<boolean> | WebTransform
+type IsObjectStdioSingleOption<StdioSingleOptionType extends StdioSingleOption> = StdioSingleOptionType extends {objectMode?: boolean}
 	? BooleanObjectMode<StdioSingleOptionType['objectMode']>
 	: StdioSingleOptionType extends DuplexTransform
-		? DuplexObjectMode<StdioSingleOptionType>
+		? StdioSingleOptionType['transform']['readableObjectMode']
 		: false;
 
 type BooleanObjectMode<ObjectModeOption extends boolean | undefined> = ObjectModeOption extends true ? true : false;
-
-type DuplexObjectMode<OutputOption extends DuplexTransform> = OutputOption['objectMode'] extends boolean
-	? OutputOption['objectMode']
-	: OutputOption['transform']['readableObjectMode'];

